Twitter

gaurdianaq opened the pull request.

Commit:
started twitter crawler, pulled in authorization library for requests

Commit:
Merge branch 'dev' of https://github.com/hatchways/team-ginger into twitter

Commit:
About to test twitter api

Commit:
Twitter integration seems to be working now, but for some reason some tweets are still getting truncated

Commit:
Conflict fixed

ryanarine commented on server/mentions_crawler_apis/twitter.py:
```diff
@@ -32,6 +34,44 @@ def one_day_ago():
+
+
+def month_to_num(month):
+    if month == "Jan":
+        return 1
+    elif month == "Feb":
+        return 2
+    elif month == "Mar":
+        return 3
+    elif month == "Apr":
+        return 4
+    elif month == "May":
+        return 5
+    elif month == "Jun":
+        return 6
+    elif month == "Jul":
+        return 7
+    elif month == "Aug":
+        return 8
+    elif month == "Sep":
+        return 9
+    elif month == "Oct":
+        return 10
+    elif month == "Nov":
+        return 11
+    elif month == "Dec":
+        return 12
```
I kinda prefer a more condensed way of doing this such as:
```
MONTHS = {"Jan": 1, "Feb": 2. ... "Dec": 12}
def month_to_num(month) {
    return MONTHS[month]
}
```

gaurdianaq replied:
not really that much more condensed since you still have to write out the whole dictionary, also no way to make it immutable, I mean I could stick months in the function but personally I prefer using the if statement for this, unless python has actual enums...

ryanarine committed:
Frontend Twitter changes
Made changes to compensate for the addition of twitter
-Moved the site to image mapping from DashboardBody and Dialog to MentionContainer
-Mentions now support a thumbnail if the backend supplies it otherwise the thumbnail will be the site logo
-Fixed bug where a 0 was being rendered if mention hits was equal to 0

ryanarine committed:
Merge branch 'dev' into twitter
